api/indexer/indexer.go

```diff
@@ -200,31 +223,45 @@ func (*Indexer) storeTransaction(
 		Fee:       fee,
 		Outputs:   outputs,
 		Error:     errorStr,
+		TxBytes:   txBytes,
 	}
 	storageTxBytes := storageTx.MarshalCanoto()
 
 	return batch.Put(txID[:], storageTxBytes)
 }
 
-func (i *Indexer) GetTransaction(txID ids.ID) (bool, int64, bool, fees.Dimensions, uint64, [][]byte, string, error) {
+func (i *Indexer) GetTransaction(txID ids.ID) (bool, *chain.Transaction, []byte, int64, *chain.Result, error) {
 	v, err := i.txDB.Get(txID[:])
 	if errors.Is(err, database.ErrNotFound) {
-		return false, 0, false, fees.Dimensions{}, 0, nil, "", nil
+		return false, nil, nil, 0, nil, nil
 	}
 	if err != nil {
-		return false, 0, false, fees.Dimensions{}, 0, nil, "", err
+		return false, nil, nil, 0, nil, err
 	}
 
 	storageTx := storageTx{}
 	if err := storageTx.UnmarshalCanoto(v); err != nil {
-		return false, 0, false, fees.Dimensions{}, 0, nil, "", fmt.Errorf("failed to unmarshal storage tx %s: %w", txID, err)
+		return false, nil, nil, 0, nil, fmt.Errorf("failed to unmarshal storage tx %s: %w", txID, err)
 	}
 
 	unpackedUnits, err := fees.UnpackDimensions(storageTx.Units)
 	if err != nil {
-		return false, 0, false, fees.Dimensions{}, 0, nil, "", fmt.Errorf("failed to unpack units for storage tx %s: %w", txID, err)
+		return false, nil, nil, 0, nil, fmt.Errorf("failed to unpack units for storage tx %s: %w", txID, err)
+	}
+
+	p := codec.NewReader(storageTx.TxBytes, consts.NetworkSizeLimit)
+	tx, err := chain.UnmarshalTx(p, i.parser.ActionCodec(), i.parser.AuthCodec())
+	if err != nil {
+		return false, nil, nil, 0, nil, fmt.Errorf("failed to unmarshal tx %s: %w", txID, err)
+	}
+	result := &chain.Result{
+		Success: storageTx.Success,
+		Error:   []byte(storageTx.Error),
+		Outputs: storageTx.Outputs,
+		Units:   unpackedUnits,
+		Fee:     storageTx.Fee,
 	}
-	return true, storageTx.Timestamp, storageTx.Success, unpackedUnits, storageTx.Fee, storageTx.Outputs, storageTx.Error, nil
+	return true, tx, storageTx.TxBytes, storageTx.Timestamp, result, nil
```

Comment:
Can we return just the transaction instead of `storageTx.TxBytes` separately since a client can already retrieve the bytes from the `*chain.Transaction` type